File: PracticasJava/Practica7/src/Juego.java
import java.util.Random;
import java.util.Scanner;
public class Juego {
	protected int numeroDeVidas;
	protected int record;
    public Juego(int numeroDeVidas, int record) {
        this.numeroDeVidas = numeroDeVidas;
        this.record = record;
        
    }
    public Juego(int numeroDeVidas) {
    	this.numeroDeVidas = numeroDeVidas;
        this.record = 0;
    }
    
    public void reiniciaPartida() {
    	this.numeroDeVidas = getNumeroDeVidas();
    	System.out.println("Partida reiniciada tiene "+ numeroDeVidas+ " vidas");
    }
    public void actualizaRecord() {
    		this.record = this.record+1;
    		System.out.println("Su record ha sido actualizado " + record);
    }
    public boolean quitaVida() {
    	if(getNumeroDeVidas()>0) {	
    		this.numeroDeVidas = this.numeroDeVidas-1;
    		return true;
    	} else {
    		return false;
    	}
    }
    public int getNumeroDeVidas() {
    	return numeroDeVidas;
    }
    public void setNumeroDeVidas(int numeroDeVidas) {
    	this.numeroDeVidas = numeroDeVidas;
    }
    public int getRecord() {
    	return record;
    }
    public void setRecord(int record) {
    	this.record= record;
    }  
}

File: PracticasPython/Practica7/Juego.py
import random
class Juego:
    def __init__(self, numeroDeVidas, record):
        self.numeroDeVidas = numeroDeVidas
        self.record = record
    def reiniciaPartida(self):
        self.numeroDeVidas = self.get_numeroDeVidas()
        print(f"Partida reiniciada {self.numeroDeVidas} vidas")
    def actualizaRecord(self):
        self.record = self.record +1 
        print(f"El record ha sido actualizado: {self.record}")
    def quitaVida(self):
        if self.get_numeroDeVidas() > 0:
            self.numeroDeVidas -= 1
            return True
        else:
            return False
    def get_record(self):
        return self.record
    def set_record(self, record):
        self.record = record
    def get_numeroDeVidas(self):
        return self.numeroDeVidas
    def set_numeroDeVidas(self, numeroDeVidas):
        self.numeroDeVidas = numeroDeVidas